pennylane/transforms/control.py

```diff
@@ -45,9 +47,28 @@ def expand_with_control(tape, control_wire):
                 # pylint: disable=protected-access
                 op._controlled(control_wire)
             else:
-                tmp_tape = op.expand()
-                tmp_tape = expand_with_control(tmp_tape, control_wire)
-                requeue_ops_in_tape(tmp_tape)
+                try:
+                    # Attempt to decompose the operation and apply
+                    # controls to each gate in the decomposition.
+                    tmp_tape = op.expand()

```

Comment:
if `tape` is a tape-of-tapes, would we ever hit the `except NotImplementedError`? I guess not, since the sub-tapes will have an expand method?

Reply:
Yes, this should already be the case, since templates are now operations, and they return tapes of tapes 🤔